### ISSUE TITLE
feat: implement 0xF0 - CREATE

### ISSUE THREAD

**Issue by TotalPizza** [sayajin-labs/kakarot#253]
### Since

Frontier

### Group

System Operations

### Description

0xF0 - CREATE

**TL;DR**
Allows for deployed contract code to trigger the creation of a new contract.

**More specifically:**

Creates a new contract. Enters a new sub [context](https://www.evm.codes/about) of the calculated destination address and executes the provided initialisation code, then resumes the current context.

Should deployment succeed, the new account's [code](https://www.evm.codes/about) is set to the [return data](https://www.evm.codes/about) resulting from executing the initialisation code.

The destination address is calculated as the rightmost 20 bytes (160 bits) of the Keccak-256 hash of the rlp encoding of the sender address followed by its nonce.

## Stack Input
1.) value: value in [wei](https://www.investopedia.com/terms/w/wei.asp) to send to the new account.
2.) offset: byte offset in the [memory](https://www.evm.codes/about) in bytes, the initialisation code for the new account.
3.) size: byte size to copy (size of the initialisation code).

## Stack Output

1.) address: the address of the deployed contract, 0 if the deployment failed.

## Error Cases
- Not enough gas.
- Not enough values on the stack.
- The current execution context is from a [STATICCALL](https://www.evm.codes/#FA) (since Byzantium fork).

**Comment by TotalPizza** [comment]
I'd like to start concepting the implementation on this one.

**Comment by TotalPizza** [comment]
Current blocker is the need for an [rlp](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/) encoding library in order to generate eth addresses in the exact same way as on evm ethereum.

**Comment by liamzebedee** [comment]
@TotalPizza you might be encouraged to find @maxgillet has written some RLP decoding utils here - https://github.com/maxgillett/stark-attestations/blob/master/programs/lib/storage_verification/extract_from_rlp.cairo.

**Comment by TotalPizza** [comment]
Implementation of CREATE can be further discussed in the draft PR:
https://github.com/sayajin-labs/kakarot/pull/293

**Comment by TotalPizza** [comment]
> @TotalPizza you might be encouraged to find @MaxGillet has written some RLP decoding utils here - https://github.com/maxgillett/stark-attestations/blob/master/programs/lib/storage_verification/extract_from_rlp.cairo.

WOAAA thanx dude.
Haven't looked into it yet, but hopefully it'll mean I don't have to write it from scratch!

**Comment by TotalPizza** [comment]
This issue is free to grab by someone else,
Unfortunately I cannot continue working on this.

**Comment by jobez** [comment]
I can give this a go.
